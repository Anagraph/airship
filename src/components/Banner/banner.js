import styled from 'styled-components';
import PropTypes from 'prop-types';
import { theme } from '../../constants';

const BannerIcon = styled.div`
  display: flex;
  flex: 0 0 auto;
`;
BannerIcon.displayName = 'Banner.Handle';

const BannerContent = styled.div`
  flex: 1;
  padding: 0 8px;
`;
BannerContent.displayName = 'Banner.Content';

const Banner = styled.div`
  align-items: center;
  background-color: ${props => props.color || props.theme.ui04};
  border-radius: 0;
  box-sizing: border-box;
  display: flex;
  padding: 12px;
  width: 100%;
`;

Banner.propTypes = {
  color: PropTypes.string,
};

Banner.defaultProps = {
  theme,
};

Banner.Icon = BannerIcon;
Banner.Content = BannerContent;

export default Banner;
