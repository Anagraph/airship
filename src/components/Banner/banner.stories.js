import React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import Banner from './banner';
import Icon from '../Icon/icon';
import Text from '../Typography/text';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  ui04: '#333',
  type01: 'white',
};

storiesOf('Banner', module)
  .add('Default', () => (
    <Banner>
      <Banner.Icon>
        <Icon icon="alert_fill" size={12} />
      </Banner.Icon>
      <Banner.Content>
        <Text>
          An error has ocurred. Could not retreive customer information
        </Text>
      </Banner.Content>
    </Banner>
  ))
  .add('With custom color', () => (
    <Banner color="#4f1091">
      <Banner.Icon>
        <Icon icon="alert_fill" color="#fff" size={12} />
      </Banner.Icon>
      <Banner.Content>
        <Text color="#fff">
          An error has ocurred. Could not retreive customer information
        </Text>
      </Banner.Content>
    </Banner>
  ))
  .add('With custom theme', () => (
    <ThemeProvider theme={CUSTOM_THEME}>
      <Banner color="#333">
        <Banner.Icon>
          <Icon icon="alert_fill" color="#fff" size={12} />
        </Banner.Icon>
        <Banner.Content>
          <Text>
            An error has ocurred. Could not retreive customer information
          </Text>
        </Banner.Content>
      </Banner>
    </ThemeProvider>
  ));
