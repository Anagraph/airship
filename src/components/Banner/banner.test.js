import React from 'react';
import renderer from 'react-test-renderer';
import Banner from './banner';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(
      <Banner color="red">
        <Banner.Icon>Icon</Banner.Icon>
        <Banner.Content>Content</Banner.Content>
      </Banner>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
