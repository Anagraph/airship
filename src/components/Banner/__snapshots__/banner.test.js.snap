// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders without crashing 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 8px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: red;
  border-radius: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px;
  width: 100%;
}

<div
  className="c0"
  color="red"
>
  <div
    className="c1"
  >
    Icon
  </div>
  <div
    className="c2"
  >
    Content
  </div>
</div>
`;
